tests/test_utils/test_schemaview.py

```diff
@@ -1012,6 +1013,403 @@ def test_schemaview_enums(schema_view_no_imports: SchemaView) -> None:
+
+    :param range_id: the range file to use; must be one of RL, RI
+    :type range_id: str
+    :param tmp_path_factory: pytest fixture for generating temporary paths
+    :type tmp_path_factory: pytest.TempPathFactory
+    :return: path to the generated file
+    :rtype: Path
+    """
+    schema_yaml = RANGE_DATA["schema_path"][range_id].read_text()
+    default_range = RANGE_DATA["default"][range_id]
+    range_default_yaml = f"{schema_yaml}\n\ndefault_range: {default_range}\n"
+    return save_temp_file(range_default_yaml, f"{default_range}.yaml", tmp_path_factory)
+
+
+@pytest.fixture(scope="session")
+def range_local_default_path(tmp_path_factory: pytest.TempPathFactory) -> Path:
+    """Generate a copy of the range_local file with a default range specified and return the path."""
+    return gen_range_file_with_default(RL, tmp_path_factory)
+
+
+@pytest.fixture(scope="session")
+def range_importer_default_path(tmp_path_factory: pytest.TempPathFactory) -> Path:
+    """Generate a copy of the range importer file with a default range specified and return the path."""
+    return gen_range_file_with_default(RI, tmp_path_factory)
+
+
+def sv_range_import_whatever(
+    request: pytest.FixtureRequest, range_tuple: tuple[str, str | None, str | None]
+) -> tuple[SchemaView, tuple[str, str | None, str | None]]:
+    """Generate a schema and optionally, schemas that import that schema.
+
+    There are three potential schemas generated:
+    - the range schema, based on RANGE_DATA["schema_path"]["range_local"]; the range is referred to as local_default;
+    - the importer schema, which imports the range schema; the range referred to as importer_default;
+      the file contents are based on RANGE_DATA["schema_path"]["range_importer"]
+    - the import_importer schema, which imports the importer schema; the range is import_importer_default;
+      the schema is constructed in this function.
+
+    The range_tuple parameter specifies the default_range values for each of these schemas in the order
+    range_local, range_importer, range_import_importer.
+
+    A value of "" (EMPTY) means that there is no default_range set.
+    A value of None means that this file should not be generated.
+
+    :param request: fixture request object for retrieving fixture-related stuff
+    :type request: pytest.FixtureRequest
+    :param range_tuple: the default_range for the range schema; if the value of local_default is "", the default_range is not set.
+    :type range_tuple: tuple[str, str|None, str|None]
+    :return: a loaded SchemaView for the appropriate schema(s)
+    :rtype: SchemaView
+    """
+    (local_default, importer_default, import_importer_default) = range_tuple
+
+    if importer_default is None and import_importer_default is not None:
+        err_msg = "If the importer is not used, any import_importer value other than None is not valid"
+        raise ValueError(err_msg)
```

Comment:
This will never happen for any of the RANGE_TUPLES combinations, but always worth adding error handling _just in case_.